### PR TITLE
Improve documentation of 0.8.x -> 0.9.0 upgrade

### PR DESCRIPTION
**Pull request by Kixunil**
This should hopefully make the major changes more visible and avoid
surprises.

Assumes #477 will get merged (in other words, don't merge before it)

Since this changes formatting & release notes significantly it felt more appropriate to open a separate PR.

I'm also thinking usage.md is quite large now, maybe split it up?

### CODE REVIEW

**Comment by MaxHillebrand** [comment]
Looks good, it's understandable.

File: RELEASE-NOTES.md [hunk 1]
```diff
@@ -1,9 +1,22 @@
...
+* Database schema changed - this will cause **reindex after upgrade**.
+* We now use **bitcoin p2p protocol** to fetch blocks - some configurations may not work.
+
+See [upgrading](doc/usage.md#upgrading) section of our docs to learn more.
+
+Full list of changes:
+
 * Fix incorrect ordering of same-block transactions (#297)
 * Change DB index format and use Zstd compression (instead of Snappy)
+* The database will be reindexed automatically when it encounters old version
```
Comment (romanz) [inline RELEASE-NOTES.md:16]:
```suggestion
* The database will be reindexed automatically when it encounters old version (#567)
```

**Review by romanz — APPROVED** [review]
Looks great, thanks!

File: RELEASE-NOTES.md [hunk 2]
```diff
@@ -1,9 +1,22 @@
...
 * Don't use bitcoind JSON RPC for fetching blocks (#373)
-* Use p2p for block fetching (instead of reading `blk*.dat` files)
+* Use p2p for block fetching only.
+  This is safer than reading `blk*dat` files and faster than Json RPC.
```
Comment (romanz) [inline RELEASE-NOTES.md:19]:
```suggestion
  This is safer than reading `blk*dat` files and faster than JSON RPC.
```

**Comment by romanz** [comment]
> I'm also thinking usage.md is quite large now, maybe split it up?

Sounds good, can be done also after the `-rc1` release :)

**Comment by Kixunil** [comment]
OK, will split after rc1.

**Review by romanz — APPROVED** [review]
LGTM :)

**Comment by Kixunil** [comment]
Because of 577f54195b8c21f63fc4abec3c9364ddd1a73692 this can be merged now.

**Comment by romanz** [comment]
Merged, many thanks!

**Pull request merged**